app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -0,0 +1,195 @@
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(ViewUtils.setGone(binding.commentsToolbar.commentButton))
+        viewModel.outputs.showCommentDialog()
+            .filter { it != null }
+            .map { it.first }
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { showCommentDialog(it) }
+        alertDialog
+            .compose(Transformers.takeWhen(viewModel.outputs.dismissCommentDialog()))
+            .observeOn(AndroidSchedulers.mainThread())
+            .compose(bindToLifecycle())
+            .subscribe { dismissCommentDialog(it) }
+        lifecycle()
+            .compose(Transformers.combineLatestPair(alertDialog))
+            .filter { it.first == ActivityEvent.DESTROY }
+            .map { it.second }
+            .observeOn(AndroidSchedulers.mainThread())
+            // NB: We dont want to bind to lifecycle because we want the destroy event.
+            // .compose(bindToLifecycle())
+            .take(1)
+            .subscribe { dismissCommentDialog(it) }
+        toastMessages()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(ViewUtils.showToast(this))
+
+        findViewById<View>(R.id.project_context_view)?.setOnClickListener { projectContextViewClick() }
+        findViewById<View>(R.id.comment_button)?.setOnClickListener { commentButtonClicked() }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        recyclerViewPaginator?.stop()
+        binding.commentsRecyclerView.adapter = null
+    }
+
+    private fun projectContextViewClick() {
+        back()
+    }
+
+    private fun commentsLogin() {
+        val intent = Intent(this, LoginToutActivity::class.java)
+            .putExtra(IntentKey.LOGIN_REASON, LoginReason.COMMENT_FEED)
+        startActivityForResult(intent, ActivityRequestCodes.LOGIN_FLOW)
+    }
+
+    private fun commentButtonClicked() {
+        viewModel.inputs.commentButtonClicked()
+    }
+
+    private fun dismissCommentDialog(dialog: AlertDialog?) {
+        dialog?.dismiss()
+    }
+
+    private fun showCommentDialog(project: Project) {
+        val commentDialog = AlertDialog.Builder(this)
+            .setView(R.layout.comment_dialog)
+            .create()
+        commentDialog.show()
+        commentDialog.window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE)
+
+        /* Toolbar UI actions */
+        val projectNameTextView = commentDialog.findViewById<TextView>(R.id.comment_project_name)
+        projectNameTextView?.text = project.name()
+
+        // Handle cancel-click region outside of dialog modal.
+        commentDialog.setOnCancelListener { dialogInterface: DialogInterface -> viewModel.inputs.commentDialogDismissed() }
+        alertDialog.onNext(commentDialog)
+    }
+
+    fun setPostButtonEnabled(postCommentButton: TextView?, enabled: Boolean) {
```

Comment:
nit picky detail, inline function`fun setPostButtonEnabled(postCommentButton: TextView?, enabled: Boolean) = postCommentButton?.let { it.isEnabled = enabled }`